sdk/nodejs/introspector/test/analysis.spec.ts

```diff
@@ -0,0 +1,331 @@
+      classes: [
+        {
+          name: "Alpine",
+          doc: "Alpine module",
+          properties: [
+            {
+              name: "packages",
+              typeName: "string[]",
+              doc: "packages to install",
+            },
+            {
+              name: "ctr",
+              typeName: "Container",
+              doc: "",
+            },
+          ],
+          methods: [
+            {
+              name: "base",
+              returnType: "Alpine",
+              doc: "Returns a base Alpine container",
+              params: [
+                {
+                  name: "version",
+                  typeName: "string",
+                  doc: "version to use (default to: 3.16.2)",
+                  optional: true,
+                },
+              ],
+            },
+            {
+              name: "install",
+              returnType: "Alpine",
+              doc: "",
+              params: [
+                {
+                  name: "pkgs",
+                  typeName: "string[]",
+                  doc: "",
+                  optional: false,
+                },
+              ],
+            },
+            {
+              name: "exec",
+              returnType: "string",
+              doc: "",
+              params: [
+                {
+                  name: "cmd",
+                  typeName: "string[]",
+                  doc: "",
+                  optional: false,
+                },
+              ],
+            },
+          ],
+        },
+      ],
+      functions: [],
+    }
+
+    assert.deepEqual(result, expected)
+  })
+
+  it("Should detect optional parameters of a method", async function () {
+    const files = await listFiles(`${rootDirectory}/optionalParameter`)
+
+    const result = analysis(files)
+    const expected: Metadata = {
+      classes: [
+        {
+          name: "HelloWorld",
+          doc: "HelloWorld class",
+          properties: [],
+          methods: [
+            {
+              name: "helloWorld",
+              returnType: "string",
+              doc: "",
+              params: [
+                {
+                  name: "name",
+                  typeName: "string",
+                  doc: "",
+                  optional: true,
+                },
+              ],
+            },
+            {
+              name: "isTrue",
+              returnType: "boolean",
+              doc: "",
+              params: [
+                {
+                  name: "value",
+                  typeName: "boolean",
+                  doc: "",
+                  optional: false,
+                },
+              ],
+            },
+            {
+              name: "add",
+              returnType: "number",
+              doc: "",
+              params: [
+                {
+                  name: "a",
+                  typeName: "number",
+                  doc: "",
+                  optional: true,
+                },
```

Comment:
Correctly identified as optional, but could add the defaultValue here.

Reply:
Why? It is not supported by GraphQL asap?

Reply:
Resolved in [27477bc](https://github.com/dagger/dagger/pull/6159/commits/27477bc051006373f25a0ad00364ef1e6bc95bea)